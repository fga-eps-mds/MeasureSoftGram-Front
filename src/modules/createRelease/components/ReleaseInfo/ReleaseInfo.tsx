import React from 'react';

import { Box, Checkbox, FormControlLabel, FormGroup, TextField } from '@mui/material';
import { useCreateReleaseContext } from '@modules/createRelease/context/useCreateRelease';
import undelineRemover from '@utils/undelineRemover';
import * as Styles from './styles';
import Skeleton from './Skeleton';

function ReleaseInfo() {
  const { releaseInfoForm, preConfigCharacteristics, handleChangeForm, handleSelectCharacteristics } =
    useCreateReleaseContext();

  if (!preConfigCharacteristics) return <Skeleton />;

  const { endDate, name, characteristics, startDate } = releaseInfoForm;

  return (
    <>
      <Styles.Header>
        <h1 style={{ color: '#33568E', fontWeight: '500' }}>Planejar Release</h1>
        <p>
          Na etapa inicial do planejamento da release é possível definir seu nome e as datas de início e fim, bem como
          selecionar as características a serem observadas durante o período planejado.
        </p>
      </Styles.Header>

      <Styles.Body>
        <Box>
          <h2 style={{ color: '#538BA3', fontWeight: '500' }}>Configurações básicas</h2>

          <TextField
            label="Apelido da release"
            style={{ marginBottom: '24px' }}
            value={name}
            required
            onChange={(e) => {
              handleChangeForm('name', e.target.value);
            }}
            inputProps={{
              'data-testid': 'apelido-release'
            }}
            fullWidth
          />

          <Box display="flex">
            <TextField
              label="Data de ínicio da release"
              type="date"
              required
              value={startDate}
              style={{ marginRight: '16px' }}
              onChange={(e) => {
                handleChangeForm('startDate', e.target.value);
              }}
              inputProps={{
                'data-testid': 'inicio-release'
              }}
              sx={{ flex: 1 }}
            />

            <TextField
              label="Data de fim da release"
              type="date"
              required
              value={endDate}
              onChange={(e) => {
                handleChangeForm('endDate', e.target.value);
              }}
              inputProps={{
                'data-testid': 'fim-release'
              }}
              sx={{ flex: 1 }}
            />
          </Box>
        </Box>

        <Box>
          <h2 style={{ color: '#538BA3', fontWeight: '500' }}>Características a serem observadas no release</h2>

          <FormGroup onChange={(e) => handleSelectCharacteristics((e.target as HTMLInputElement).value)}>
            {preConfigCharacteristics.map((item) => (
              <FormControlLabel
                key={item}
                control={
                  <Checkbox
                    value={item}
                    name={item}
                    id={item}
                    checked={characteristics?.includes(item)}
                    inputProps={{
                      'data-testid': 'characteristic-release'
                    }}
                    color="secondary"
                  />
                }
                label={undelineRemover(item)}
              />
            ))}
          </FormGroup>
        </Box>
      </Styles.Body>
    </>
  );
}

export default ReleaseInfo;
