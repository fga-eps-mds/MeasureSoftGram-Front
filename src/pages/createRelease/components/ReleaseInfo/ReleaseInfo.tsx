import React from 'react';

import { Box, Checkbox, FormControlLabel, FormGroup, TextField } from '@mui/material';
import { useCreateReleaseContext } from '@pages/createRelease/context/useCreateRelease';
import undelineRemover from '@utils/undelineRemover';
import * as Styles from './styles';
import Skeleton from './Skeleton';

function ReleaseInfo() {
  const { releaseInfoForm, preConfigCharacteristics, handleChangeForm, handleSelectCharacteristics } =
    useCreateReleaseContext();

  if (!preConfigCharacteristics) return <Skeleton />;

  const { endDate, name, characteristics, startDate } = releaseInfoForm;

  return (
    <>
      <Styles.Header>
        <h1>Definir Meta de Qualidade (planejada)</h1>
        <p>Mini explicação sobre as configurações basicas e a seleção de caracteristicas</p>
      </Styles.Header>

      <Styles.Body>
        <Box>
          <h2>Configurações básicas</h2>

          <TextField
            label="Apelido da release"
            style={{ marginBottom: '24px' }}
            value={name}
            required
            onChange={(e) => {
              handleChangeForm('name', e.target.value);
            }}
            inputProps={{
              "data-testid": "apelido-release",
            }}
            fullWidth
          />

          <Box display="flex">
            <TextField
              label="Data de ínicio da release"
              type="date"
              required
              value={startDate}
              style={{ marginRight: '16px' }}
              onChange={(e) => {
                handleChangeForm('startDate', e.target.value);
              }}
              inputProps={{
                "data-testid": "inicio-release",
              }}
              sx={{ flex: 1 }}
            />

            <TextField
              label="Data de fim da release"
              type="date"
              required
              value={endDate}
              onChange={(e) => {
                handleChangeForm('endDate', e.target.value);
              }}
              inputProps={{
                "data-testid": "fim-release",
              }}
              sx={{ flex: 1 }}
            />
          </Box>
        </Box>

        <Box>
          <h2>Caracteristicas a serem observadas no release</h2>

          <FormGroup onChange={(e) => handleSelectCharacteristics((e.target as HTMLInputElement).value)}>
            {preConfigCharacteristics.map((item) => (
              <FormControlLabel
                key={item}
                control={
                  <Checkbox
                    value={item}
                    name={item}
                    id={item}
                    checked={characteristics?.includes(item)}
                    inputProps={{
                      "data-testid": "characteristic-release",
                    }}
                  />
                }
                label={undelineRemover(item)}
              />
            ))}
          </FormGroup>
        </Box>
      </Styles.Body>
    </>
  );
}

export default ReleaseInfo;
