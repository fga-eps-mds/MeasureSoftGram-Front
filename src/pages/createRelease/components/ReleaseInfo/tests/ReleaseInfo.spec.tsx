import '@testing-library/jest-dom';

import React from 'react';
import { act, fireEvent, render } from '@testing-library/react';

import ReleaseInfo from '../ReleaseInfo';

jest.mock('@pages/createRelease/context/useCreateRelease', () => ({
  useCreateReleaseContext: () => ({
    releaseInfoForm: {
      characteristics: [ "reliability", "maintainability" ],
      endDate: "2022-09-25",
      name: "asdasd",
      startDate: "2022-09-18",
    },
    preConfigCharacteristics: ['reliability', 'maintainability'],
    handleChangeForm: () => {},
    handleSelectCharacteristics: () => {}
  })
}));

describe('<ReleaseInfo />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Snapshot', () => {
    it('Deve corresponder ao Snapshot', () => {
      const tree = render(<ReleaseInfo/>)
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Comportamento', () => {
    it('Deve preencher campos do formulario', () => {
      const { getByTestId, getAllByTestId } = render(<ReleaseInfo/>)

      act(() => {
        fireEvent.change(getByTestId("apelido-release"), { target: { value: 'aoba' } });
        fireEvent.change(getByTestId("inicio-release"), { target: { value: '2022-09-25' } });
        fireEvent.change(getByTestId("fim-release"), { target: { value: '2022-09-31' } });
        fireEvent.click(getAllByTestId("characteristic-release")[0], { target: { checked: true } });
      });

      expect(true).toBeTruthy();
    });
  });
});
