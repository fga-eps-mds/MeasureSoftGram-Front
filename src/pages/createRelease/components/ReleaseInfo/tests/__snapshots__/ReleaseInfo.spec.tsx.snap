// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReleaseInfo /> Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Header-sc-knppiq-0 hmeceq"
      >
        <h1>
          Definir Meta de Qualidade (planejada)
        </h1>
        <p>
          Mini explicação sobre as configurações basicas e a seleção de caracteristicas
        </p>
      </div>
      <div
        class="styles__Body-sc-knppiq-1 jiofxz"
      >
        <div
          class="MuiBox-root css-0"
        >
          <h2>
            Configurações básicas
          </h2>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            style="margin-bottom: 24px;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r0:"
              id=":r0:-label"
            >
              Apelido da release
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="apelido-release"
                id=":r0:"
                required=""
                type="text"
                value="asdasd"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Apelido da release
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-k008qs"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1vs1jgp-MuiFormControl-root-MuiTextField-root"
              style="margin-right: 16px;"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for=":r1:"
                id=":r1:-label"
              >
                Data de ínicio da release
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="inicio-release"
                  id=":r1:"
                  required=""
                  type="date"
                  value="2022-09-18"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1in441m"
                  >
                    <span>
                      Data de ínicio da release
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root css-1vs1jgp-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for=":r2:"
                id=":r2:-label"
              >
                Data de fim da release
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="fim-release"
                  id=":r2:"
                  required=""
                  type="date"
                  value="2022-09-25"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1in441m"
                  >
                    <span>
                      Data de fim da release
                       
                      *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <h2>
            Caracteristicas a serem observadas no release
          </h2>
          <div
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  data-testid="characteristic-release"
                  id="reliability"
                  name="reliability"
                  type="checkbox"
                  value="reliability"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                reliability
              </span>
            </label>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  checked=""
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  data-testid="characteristic-release"
                  id="maintainability"
                  name="maintainability"
                  type="checkbox"
                  value="maintainability"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                maintainability
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Header-sc-knppiq-0 hmeceq"
    >
      <h1>
        Definir Meta de Qualidade (planejada)
      </h1>
      <p>
        Mini explicação sobre as configurações basicas e a seleção de caracteristicas
      </p>
    </div>
    <div
      class="styles__Body-sc-knppiq-1 jiofxz"
    >
      <div
        class="MuiBox-root css-0"
      >
        <h2>
          Configurações básicas
        </h2>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          style="margin-bottom: 24px;"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            Apelido da release
            <span
              aria-hidden="true"
              class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="apelido-release"
              id=":r0:"
              required=""
              type="text"
              value="asdasd"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1in441m"
              >
                <span>
                  Apelido da release
                   
                  *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-k008qs"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1vs1jgp-MuiFormControl-root-MuiTextField-root"
            style="margin-right: 16px;"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r1:"
              id=":r1:-label"
            >
              Data de ínicio da release
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="inicio-release"
                id=":r1:"
                required=""
                type="date"
                value="2022-09-18"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Data de ínicio da release
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1vs1jgp-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for=":r2:"
              id=":r2:-label"
            >
              Data de fim da release
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="fim-release"
                id=":r2:"
                required=""
                type="date"
                value="2022-09-25"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Data de fim da release
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <h2>
          Caracteristicas a serem observadas no release
        </h2>
        <div
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                data-testid="characteristic-release"
                id="reliability"
                name="reliability"
                type="checkbox"
                value="reliability"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              reliability
            </span>
          </label>
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate="false"
                data-testid="characteristic-release"
                id="maintainability"
                name="maintainability"
                type="checkbox"
                value="maintainability"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              maintainability
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
